sql/scan.go

```diff
@@ -157,6 +161,10 @@ func (n *scanNode) Next() bool {
 		}
 	}
 
+	if n.limit != parser.DNull && n.renderRowIndex >= int(n.limit.(parser.DInt)) {
```

Comment:
Perhaps move this above the call to `n.initScan` so that we never even bother scanning if the limit is 0.